chrome/extension/ergo-connector/api.js

```diff
@@ -139,28 +157,31 @@ export async function connectorGetUtxos(
     throw new Error('wallet doesn\'t support IGetAllUtxos');
   }
   const utxos = await withUtxos.getAllUtxos();
-  // TODO: more intelligently choose values?
+  const spentBoxIds = pendingTxs.flatMap(pending => pending.tx.inputs.map(input => input.boxId));
+  // TODO: should we use a different coin selection algorithm besides greedy?
   let utxosToUse = [];
   if (valueExpected != null) {
     let valueAcc = new BigNumber(0);
     const target = valueToBigNumber(valueExpected);
     for (let i = 0; i < utxos.length && valueAcc.isLessThan(target); i += 1) {
```

Comment:
Why do we use custom code for this instead of the input selection that comes with sigma-rust?

Reply:
It'd end up being more work to go to/from the connector API types <-> sigma-rust wasm types, and at the time the code was first written we weren't loading the rust module, as we only need it for syncing + signing, not for the other API calls.